model.py

```diff
@@ -7953,8 +7953,8 @@ class CustomList(Base):
     id = Column(Integer, primary_key=True)
     primary_language = Column(Unicode, index=True)
     data_source_id = Column(Integer, ForeignKey('datasources.id'), index=True)
-    foreign_identifier = Column(Unicode, index=True)
-    name = Column(Unicode)
+    foreign_identifier = Column(Unicode, index=True, unique=True)
+    name = Column(Unicode, index=True, unique=True)
```

Comment:
Maybe (data_source_id, foreign_identifier) and (data_source_id, name) could each have a unique constraint? I can see several different data sources providing a list called 'Best Sellers' or some other common phrase.